docs/interop.md

```diff
@@ -0,0 +1,71 @@
+# Interoperability via Scripting
+
+Token SDK supports interoperability, cross-chain operations, via scripting. 
+It allows spending a token to a script by encoding the script in the `Owner` field of a `Token`, and the different drivers are capable of interpreting the owner as a script.
+After the ownership is assigned to a script, the script is evaluated at the time of spending the token. 
+The right to spend a token is enforced according to the conditions within the script.
+
+## HTLC
+
+HTLC (Hash Time Locked Contract) is a token transfer that use hashlocks and timelocks to require that the recipient of a token either acknowledge receiving the token prior to a deadline by generating cryptographic proof or forfeit the ability to claim the token, returning it to the sender.
+With this mechanism Token SDK supports atomic cross-chain swap of tokens. 
+
+### HTLC script
+
+The HTLC script encodes the details of the htlc, the identities of the sender and the recipient of the token, a deadline, and hashing information.
+The hashing information includes the hash itself, the hash function used (e.g., SHA-256), and the encoding (e.g., Base64).
+The hash is chosen by the sender and the recipient must provide the preimage for the transfer to happen.
+
+```go
+// Script contains the details of an htlc
+type Script struct {
+    Sender    view.Identity
+    Recipient view.Identity
+    Deadline  time.Time
+    HashInfo  HashInfo
+}
+
+// HashInfo contains the information regarding the hashing
+type HashInfo struct {
+    Hash         []byte
+    HashFunc     crypto.Hash
+    HashEncoding encoding.Encoding
+}
+```
+
+## Interoperability services
+
+The interoperability services which are responsible for assembling the token transaction, managing the transaction lifecycle, and so on, are the same as the [`Token Transaction Services`](./services.md). 
+They are located in `token/services/interop`.
+
+Other services are script specific.
+For example, the token transaction assembling service enables appending lock, claim, or reclaim actions to the token request of the transaction. All of these actions translate into a transfer action. 
```

Comment:
this is confusing when on considers the first sentence in the section and what follows. there seems to be a distinction between the services mentioned at the beginning and what comes after.